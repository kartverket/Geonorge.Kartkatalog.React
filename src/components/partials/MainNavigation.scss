@import '../../scss/abstracts/all';

:local(.mainNavigationContainer) {
    position: fixed;
    width: 100%;
    top: 0;
    z-index: 2;
    background-color: $primary-background;
    -webkit-box-shadow: 0 2px 5px rgba(0, 0, 0, .26);
    -moz-box-shadow: 0 2px 5px rgba(0, 0, 0, .26);
    box-shadow: 0 2px 5px rgba(0, 0, 0, .26);
}
:local(.breadCrumb) {
    ul {
        display: flex;
        margin-top: 20px;
        li {
            > a {
                font-weight: 400;
            }   
            padding: 0 10px 0 0;
            > svg {
                margin-left: 10px;
            }
            &:first-child {
                padding-left: 10px;
            }
        }
    }
}

:local(.menu) {
    overflow: hidden;
    width: auto;
    transition: all .3s ease-in-out;
    height: 0;
    position: absolute;
    right: 0;
    @include breakpoint(laptop) {
        right: 10%;        
    }
    padding: 0;
    border: 1px solid #e5e5e5;
    border-radius: 3px;
    top: 50px;
    border-width: 0;
    background-color: white;
    
    &:local(.open) {
        border-width: 2px;
        height: auto;
        max-height: 800px;
        padding: 5px;
               
        overflow: auto;
    } 
}


:local(.openmap) {
    display: block;
    &:hover {
        cursor: pointer;
    }
}
:local(.expandeddownload) {
    overflow: hidden;
    width: auto;
    transition: all .3s ease-in-out;
    height: 0;
    position: absolute;
    right: 20px;    
    padding: 0;
    border: 1px solid #e5e5e5;
    border-radius: 3px;
    top: 50px;
    border-width: 0;
    background-color: white;
    &:local(.open) {
        border-width: 2px;
        height: auto;
        max-height: 800px;
        padding: 10px;
        overflow: auto;
    } 
}
:local(.selectedlayers) {
    overflow: hidden;
    width: auto;
    transition: all .3s ease-in-out;
    height: 0;
    position: absolute;
    right: 20px;
    padding: 0;
    border: 1px solid #e5e5e5;
    border-radius: 3px;
    top: 50px;
    border-width: 0;
    background-color: white;
    &:local(.open) {
        border-width: 2px;
        height: auto;
        max-height: 800px;
        padding: 10px;
        overflow: auto;
    }
}
:local(.mapitems) {
    li {
        padding: 10px;
        border-bottom: 1px solid #e5e5e5;
        width: 100%;
        &:hover {
            background-color: #e5e5e5;
        }
    
    > span {
        &:hover {
            cursor: pointer;
        }
        padding-right: 20px;
    }
    > svg {
        float: right;
        &:hover {
            cursor: pointer;
        }
    }
        }
}
:local(.mapBtnlink) {
    border: none;
    background: transparent;
    text-decoration: none;
    font-size: 1.2em;
}
:local(.mainNavigation) {
    display: flex;
    flex-wrap: wrap;
    padding: 10px 20px;

    @media (min-width: 35em) {
        flex-wrap: nowrap;
    }

    :local(.logo) {
        flex: 1;
        min-width: 150px;

        @media (min-width: 35em) {
            flex: 0;
            min-width: 200px;
        }

        @media (min-width: 62em) {
            min-width: 248px;
        }

        a {
            display: inline-block;
        }

        img {
            height: 24px;
            width: 150px;

            @media (min-width: 35em) {
                height: 28px;
                width: 175px;
            }

            @media (min-width: 62em) {
                height: 37px;
                width: 233px;
            }
        }
    }

    :local(.search) {
        order: 3;
        width: 100%;
        @media (min-width: 35em) {
            flex: 1;
            order: 0;
            padding: 0 1em;
        }
    }

    

    :local(.iconButton) {
        flex: 0 1;
        display: block;
        margin-left: 12px;
        transition: all .3s ease-in-out;
        svg {
            font-size: 2.5em;
            color: $icon-enabled;
            
        }
        :local(.content) {
            color:$icon-enabled;
        }
        :local(.noContent) {
            color: $icon-disabled;
        }

        img {
            width: 37px;
            height: 37px;
            vertical-align: middle;
        }

        :local(.counter) {
            background-color: $primary-brand;
            border-radius: 50%;
            color: #fff;
            font-size: .857em;
            height: 1.5em;
            line-height: 1.5;
            margin-top: -.25em;
            margin-left: -.5em;
            position: absolute;
            text-align: center;
            width: 1.5em;
        }

    }
}
:local(.openMaplink) {
    display: block;
    text-align: center;
    font-size: 1.4em;

    > svg {
        margin-left: 15px;
        margin-top: 10px;
        position: relative;
        top: 5px;
        font-size: 1.6em;
    }
}
:local(.searchResults) {
    position: absolute;
    background: #fbfcfd;
    -webkit-box-shadow: 0 2px 5px rgba(0, 0, 0, .26);
    -moz-box-shadow: 0 2px 5px rgba(0, 0, 0, .26);
    box-shadow: 0 2px 5px rgba(0, 0, 0, .26);
    border-radius: 2px 0 0 2px;
    display: none;
    top: 37px;
    z-index: 1;

    &.active {
        display: block;
        width: 100%;
    }

    :local(.searchResultsSectionHeadingContainer) {
        padding: 4px 10px;

        :local(.searchResultsSectionHeading) {
            font-size: 1em;
            display: inline-block;
        }

        :local(.counter) {
            padding: 2px 4px;
            color: #fff;
            border-radius: 4px;
            background-color: #fe5000;
            display: inline-block;
            font-size: 1em;
            vertical-align: text-bottom;
            margin-left: 4px;

        }

        :local(.showAllButton) {
            float: right;
            cursor: pointer;
            font-size: 1em;
        }
    }

    :local(.searchResultsSection) {
        display: block;
        margin-bottom: 12px;

        &:nth-child(odd){
            background: #f5f6f7;
        }

        :local(.searchResultsItem) {
            display: block;
            font-size: .8em;
            color: #333435;
            padding: 4px 20px;
            -moz-transition: background-color .2s ease-in-out;
            -o-transition: background-color .2s ease-in-out;
            -webkit-transition: background-color .2s ease-in-out;
            transition: background-color .2s ease-in-out;
            cursor: pointer;
        }

        :local(.searchResultsItem:hover) {
            background: #f0f1f2;
        }
    }
}