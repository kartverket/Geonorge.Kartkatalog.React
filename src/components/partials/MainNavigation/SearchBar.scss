:local(.searchInput) {
    display: flex;
    height: 3.5em;
    padding-top: .75em;
    position: relative;

    @media (min-width: 35em) {
        height: 100%;
        padding-top: 0;
    }

    input {
        outline-offset: -2px;
        -webkit-appearance: none;
        background: #fff;
        border: 1px solid #dfdcda;
        border-right: 0;
        border-radius: 2px 0 0 2px;
        font-size: 1.2em;
        flex: 1;
        padding: 0 .75em;
    }

    button {
        -webkit-appearance: button;
        background: #fff;
        border: 1px solid #dfdcda;
        border-left: 0;
        border-radius: 0 2px 2px 0;
        display: inline-block;
        padding: 0 .75em;

        img {
            height: 1.544em;
            width: 1.5em;
        }
    }
}

:local(.searchResults) {
    position: absolute;
    background: #fbfcfd;
    -webkit-box-shadow: 0 2px 5px rgba(0, 0, 0, .26);
    -moz-box-shadow: 0 2px 5px rgba(0, 0, 0, .26);
    box-shadow: 0 2px 5px rgba(0, 0, 0, .26);
    border-radius: 2px 0 0 2px;
    display: none;
    top: 37px;
    z-index: 1;

    &.active {
        display: block;
        width: 100%;
    }
}
