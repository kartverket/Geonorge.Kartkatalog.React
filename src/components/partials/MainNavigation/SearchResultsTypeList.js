import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';


import style from './SearchResultsTypeList.scss';

class SearchResultsTypeList extends Component {

	constructor(props) {
		super(props);
		this.state = {
		};
	}

	renderDropdownResults() {
		if (this.props.searchResults && this.props.searchResults.Results) {
			const resultsTypeElements = this.props.searchResults.Results.map((result, i) => {
				return <div className={style.searchResultsItem} key={i}><a href={result.ShowDetailsUrl} key={i}>{result.Title}</a></div>
			})
			return resultsTypeElements;
		}
	}

	showResults() {

	}
	
	render() {
		return (
			<div className={style.searchResultsSection}>
				<div className={style.searchResultsSectionHeadingContainer}>
	                <span className={style.searchResultsSectionHeading}>{this.props.searchResultsType}</span>
	                <span className={style.counter}>{this.props.searchResults.NumFound}</span>
	                <span onClick={() => this.showResults()} className={style.showAllButton}>
	                  Vis alle
	                </span>
	            </div>
				{this.renderDropdownResults()}
			</div>
			)
	}
}

SearchResultsTypeList.propTypes = {
	searchResultsType: PropTypes.string.isRequired,
	searchResults: PropTypes.object.isRequired
}

export default connect(null)(SearchResultsTypeList);