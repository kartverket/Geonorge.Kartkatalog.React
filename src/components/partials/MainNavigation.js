import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom'
import { fetchMapItems } from '../../actions/MapItemActions'

import SearchBar from './MainNavigation/SearchBar';

import logo from '../../images/svg/geonorge-navbar-logo.svg';
import style from './MainNavigation.scss'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

class MainNavigation extends Component {

	componentWillMount() {
	    this.props.fetchMapItems();
	}
	
	render() {
		return (
			<div className={style.mainNavigationContainer}>
				<div className={style.mainNavigation + ' container'}>
					<Link to={'/'}>
                        <div className={style.logo}>
                            <img src={logo}></img>
                        </div>
                    </Link>
                    <div className={style.search}>
						<SearchBar />
                    </div>
                    <span className={style.iconButton} style={{display: "none"}}>
                        <span className={style.counter}>12</span>
                        <img src={require('../../images/svg/download-icon.svg')}></img>
                    </span>
                    <Link to={'/kart'}>
                        <span className={style.iconButton}>
                            <span className={style.counter}>{this.props.mapItems.length}</span>
                            <FontAwesomeIcon icon={'map-marker-alt'} className={this.props.mapItems.length > 0 ? style.content : style.fisk}/>
                        </span>
                    </Link>
				</div>
			</div>
			)
	}
}

MainNavigation.propTypes = {
	fetchMapItems: PropTypes.func.isRequired,
	mapItems: PropTypes.array.isRequired
}

const mapStateToProps = state => ({
	mapItems: state.mapItems
});

export default connect(mapStateToProps, { fetchMapItems })(MainNavigation);
