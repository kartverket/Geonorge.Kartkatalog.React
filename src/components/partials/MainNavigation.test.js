import React from 'react';
import { shallow } from "enzyme";
import { MainNavigation } from './MainNavigation';

import SearchBar from './MainNavigation/SearchBar';

import style from './MainNavigation.scss'

function setup() {
  const props = {
    fetchMapItems: jest.fn(),
    removeMapItem: jest.fn(),
    fetchItemsToDownload: jest.fn(),
    removeItemSelectedForDownload: jest.fn(),
    getDownloadItemMetadata: jest.fn(),
    fetchGeonorgeMenu: jest.fn(),
    getGeonorgeLogo: jest.fn(),
    autoAddItemFromLocalStorage: jest.fn(),
    getResource: jest.fn(),
    updateOidcCookie: jest.fn(),
    geonorgeMenu: [],
    mapItems: [],
    itemsToDownload: []
  }

  const wrapper = shallow(<MainNavigation {...props} />)

  return {
    props,
    wrapper
  }
}

describe('MainNavigation', () => {
  it('should render self and subcomponents', () => {
    const { wrapper } = setup()
    expect(wrapper.hasClass(style.mainNavigationContainer)).toBe(true);
    expect(wrapper.find('img').length).toBe(1);
    expect(wrapper.find(SearchBar).length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  })

  it('should include 1 SearchBar component', () => {
    const { wrapper } = setup()
    expect(wrapper.find(SearchBar).length).toBe(1);
  })

  it('should call fetchMapItems on mount', () => {
    const { props } = setup()
    expect(props.fetchMapItems.mock.calls.length).toBe(1)
  })

  it('should call fetchItemsToDownload on mount', () => {
    const { props } = setup()
    expect(props.fetchItemsToDownload.mock.calls.length).toBe(1)
  })
})
