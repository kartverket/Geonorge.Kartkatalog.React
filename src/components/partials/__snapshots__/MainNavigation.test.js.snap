// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNavigation should render self and subcomponents 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainNavigation
    fetchItemsToDownload={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    fetchMapItems={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    itemsToDownload={Array []}
    mapItems={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="mainNavigation container"
      >
        <Link
          replace={false}
          to="/"
        >
          <div
            className="logo"
          >
            <img
              alt="Geonorge logo"
              src="test-file-stub"
            />
          </div>
        </Link>
        <div
          className="search"
        >
          <ErrorBoundary>
            <Connect(SearchBar) />
          </ErrorBoundary>
        </div>
        <Link
          replace={false}
          to="/kart"
        >
          <span
            className="iconButton"
          >
            <span
              className="counter"
            >
              0
            </span>
            <FontAwesomeIcon
              border={false}
              className="content1"
              fixedWidth={false}
              flip={null}
              icon="map-marker-alt"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </span>
        </Link>
        <div
          className="openmap"
          onClick={[Function]}
        >
          <span
            className="iconButton"
          >
            <span
              className="counter"
            >
              0
            </span>
            <FontAwesomeIcon
              border={false}
              className="content1"
              fixedWidth={false}
              flip={null}
              icon="cloud-download"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </span>
        </div>
        <div
          className="expandeddownload"
        >
          <Link
            replace={false}
            to="/nedlasting"
          >
            Åpne nedlastinger
          </Link>
          Liste over nedlastinger
        </div>
      </div>,
      "className": "mainNavigationContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/"
          >
            <div
              className="logo"
            >
              <img
                alt="Geonorge logo"
                src="test-file-stub"
              />
            </div>
          </Link>,
          <div
            className="search"
          >
            <ErrorBoundary>
              <Connect(SearchBar) />
            </ErrorBoundary>
          </div>,
          <Link
            replace={false}
            to="/kart"
          >
            <span
              className="iconButton"
            >
              <span
                className="counter"
              >
                0
              </span>
              <FontAwesomeIcon
                border={false}
                className="content1"
                fixedWidth={false}
                flip={null}
                icon="map-marker-alt"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </span>
          </Link>,
          <div
            className="openmap"
            onClick={[Function]}
          >
            <span
              className="iconButton"
            >
              <span
                className="counter"
              >
                0
              </span>
              <FontAwesomeIcon
                border={false}
                className="content1"
                fixedWidth={false}
                flip={null}
                icon="cloud-download"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </span>
          </div>,
          <div
            className="expandeddownload"
          >
            <Link
              replace={false}
              to="/nedlasting"
            >
              Åpne nedlastinger
            </Link>
            Liste over nedlastinger
          </div>,
        ],
        "className": "mainNavigation container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="logo"
            >
              <img
                alt="Geonorge logo"
                src="test-file-stub"
              />
            </div>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="Geonorge logo"
                src="test-file-stub"
              />,
              "className": "logo",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Geonorge logo",
                "src": "test-file-stub",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ErrorBoundary>
              <Connect(SearchBar) />
            </ErrorBoundary>,
            "className": "search",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(SearchBar) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <span
              className="iconButton"
            >
              <span
                className="counter"
              >
                0
              </span>
              <FontAwesomeIcon
                border={false}
                className="content1"
                fixedWidth={false}
                flip={null}
                icon="map-marker-alt"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </span>,
            "replace": false,
            "to": "/kart",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="counter"
                >
                  0
                </span>,
                <FontAwesomeIcon
                  border={false}
                  className="content1"
                  fixedWidth={false}
                  flip={null}
                  icon="map-marker-alt"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />,
              ],
              "className": "iconButton",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                  "className": "counter",
                },
                "ref": null,
                "rendered": 0,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": "map-alt",
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "content1",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "map-marker-alt",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "span",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="iconButton"
            >
              <span
                className="counter"
              >
                0
              </span>
              <FontAwesomeIcon
                border={false}
                className="content1"
                fixedWidth={false}
                flip={null}
                icon="cloud-download"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </span>,
            "className": "openmap",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="counter"
                >
                  0
                </span>,
                <FontAwesomeIcon
                  border={false}
                  className="content1"
                  fixedWidth={false}
                  flip={null}
                  icon="cloud-download"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />,
              ],
              "className": "iconButton",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                  "className": "counter",
                },
                "ref": null,
                "rendered": 0,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "content1",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "cloud-download",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/nedlasting"
              >
                Åpne nedlastinger
              </Link>,
              "Liste over nedlastinger",
            ],
            "className": "expandeddownload",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Åpne nedlastinger",
                "replace": false,
                "to": "/nedlasting",
              },
              "ref": null,
              "rendered": "Åpne nedlastinger",
              "type": [Function],
            },
            "Liste over nedlastinger",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="mainNavigation container"
        >
          <Link
            replace={false}
            to="/"
          >
            <div
              className="logo"
            >
              <img
                alt="Geonorge logo"
                src="test-file-stub"
              />
            </div>
          </Link>
          <div
            className="search"
          >
            <ErrorBoundary>
              <Connect(SearchBar) />
            </ErrorBoundary>
          </div>
          <Link
            replace={false}
            to="/kart"
          >
            <span
              className="iconButton"
            >
              <span
                className="counter"
              >
                0
              </span>
              <FontAwesomeIcon
                border={false}
                className="content1"
                fixedWidth={false}
                flip={null}
                icon="map-marker-alt"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </span>
          </Link>
          <div
            className="openmap"
            onClick={[Function]}
          >
            <span
              className="iconButton"
            >
              <span
                className="counter"
              >
                0
              </span>
              <FontAwesomeIcon
                border={false}
                className="content1"
                fixedWidth={false}
                flip={null}
                icon="cloud-download"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </span>
          </div>
          <div
            className="expandeddownload"
          >
            <Link
              replace={false}
              to="/nedlasting"
            >
              Åpne nedlastinger
            </Link>
            Liste over nedlastinger
          </div>
        </div>,
        "className": "mainNavigationContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/"
            >
              <div
                className="logo"
              >
                <img
                  alt="Geonorge logo"
                  src="test-file-stub"
                />
              </div>
            </Link>,
            <div
              className="search"
            >
              <ErrorBoundary>
                <Connect(SearchBar) />
              </ErrorBoundary>
            </div>,
            <Link
              replace={false}
              to="/kart"
            >
              <span
                className="iconButton"
              >
                <span
                  className="counter"
                >
                  0
                </span>
                <FontAwesomeIcon
                  border={false}
                  className="content1"
                  fixedWidth={false}
                  flip={null}
                  icon="map-marker-alt"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
              </span>
            </Link>,
            <div
              className="openmap"
              onClick={[Function]}
            >
              <span
                className="iconButton"
              >
                <span
                  className="counter"
                >
                  0
                </span>
                <FontAwesomeIcon
                  border={false}
                  className="content1"
                  fixedWidth={false}
                  flip={null}
                  icon="cloud-download"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
              </span>
            </div>,
            <div
              className="expandeddownload"
            >
              <Link
                replace={false}
                to="/nedlasting"
              >
                Åpne nedlastinger
              </Link>
              Liste over nedlastinger
            </div>,
          ],
          "className": "mainNavigation container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="logo"
              >
                <img
                  alt="Geonorge logo"
                  src="test-file-stub"
                />
              </div>,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="Geonorge logo"
                  src="test-file-stub"
                />,
                "className": "logo",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Geonorge logo",
                  "src": "test-file-stub",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ErrorBoundary>
                <Connect(SearchBar) />
              </ErrorBoundary>,
              "className": "search",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Connect(SearchBar) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <span
                className="iconButton"
              >
                <span
                  className="counter"
                >
                  0
                </span>
                <FontAwesomeIcon
                  border={false}
                  className="content1"
                  fixedWidth={false}
                  flip={null}
                  icon="map-marker-alt"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
              </span>,
              "replace": false,
              "to": "/kart",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="counter"
                  >
                    0
                  </span>,
                  <FontAwesomeIcon
                    border={false}
                    className="content1"
                    fixedWidth={false}
                    flip={null}
                    icon="map-marker-alt"
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  />,
                ],
                "className": "iconButton",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0,
                    "className": "counter",
                  },
                  "ref": null,
                  "rendered": 0,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": "map-alt",
                  "nodeType": "function",
                  "props": Object {
                    "border": false,
                    "className": "content1",
                    "fixedWidth": false,
                    "flip": null,
                    "icon": "map-marker-alt",
                    "inverse": false,
                    "listItem": false,
                    "mask": null,
                    "pull": null,
                    "pulse": false,
                    "rotation": null,
                    "size": null,
                    "spin": false,
                    "symbol": false,
                    "title": "",
                    "transform": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "span",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="iconButton"
              >
                <span
                  className="counter"
                >
                  0
                </span>
                <FontAwesomeIcon
                  border={false}
                  className="content1"
                  fixedWidth={false}
                  flip={null}
                  icon="cloud-download"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
              </span>,
              "className": "openmap",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="counter"
                  >
                    0
                  </span>,
                  <FontAwesomeIcon
                    border={false}
                    className="content1"
                    fixedWidth={false}
                    flip={null}
                    icon="cloud-download"
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  />,
                ],
                "className": "iconButton",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0,
                    "className": "counter",
                  },
                  "ref": null,
                  "rendered": 0,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "border": false,
                    "className": "content1",
                    "fixedWidth": false,
                    "flip": null,
                    "icon": "cloud-download",
                    "inverse": false,
                    "listItem": false,
                    "mask": null,
                    "pull": null,
                    "pulse": false,
                    "rotation": null,
                    "size": null,
                    "spin": false,
                    "symbol": false,
                    "title": "",
                    "transform": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  replace={false}
                  to="/nedlasting"
                >
                  Åpne nedlastinger
                </Link>,
                "Liste over nedlastinger",
              ],
              "className": "expandeddownload",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Åpne nedlastinger",
                  "replace": false,
                  "to": "/nedlasting",
                },
                "ref": null,
                "rendered": "Åpne nedlastinger",
                "type": [Function],
              },
              "Liste over nedlastinger",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
